tests/scripts/thread-cert/Cert_5_1_04_RouterAddressReallocation.py

```diff
@@ -72,124 +98,172 @@ def test(self):
-        msg.assertSentWithHopLimit(255)
-        msg.assertSentToDestinationAddress("ff02::2")
-        msg.assertMleMessageContainsTlv(mle.Mode)
-        msg.assertMleMessageContainsTlv(mle.Challenge)
-        msg.assertMleMessageContainsTlv(mle.ScanMask)
-        msg.assertMleMessageContainsTlv(mle.Version)
-
-        scan_mask_tlv = msg.get_mle_message_tlv(mle.ScanMask)
-        self.assertEqual(1, scan_mask_tlv.router)
-        self.assertEqual(0, scan_mask_tlv.end_device)
-
-        # 8 - Router2
-        router2_messages.next_mle_message(mle.CommandType.PARENT_REQUEST)
-        router2_messages.next_mle_message(mle.CommandType.CHILD_ID_REQUEST)
-
-        # 9 - Router1
-        msg = router1_messages.next_mle_message(mle.CommandType.PARENT_RESPONSE)
-        msg.assertSentToNode(self.nodes[ROUTER2])
-        msg.assertMleMessageContainsTlv(mle.SourceAddress)
-        msg.assertMleMessageContainsTlv(mle.LeaderData)
-        msg.assertMleMessageContainsTlv(mle.LinkLayerFrameCounter)
-        msg.assertMleMessageContainsOptionalTlv(mle.MleFrameCounter)
-        msg.assertMleMessageContainsTlv(mle.Response)
-        msg.assertMleMessageContainsTlv(mle.Challenge)
-        msg.assertMleMessageContainsTlv(mle.LinkMargin)
-        msg.assertMleMessageContainsTlv(mle.Connectivity)
-        msg.assertMleMessageContainsTlv(mle.Version)
-
-        msg = router1_messages.next_mle_message(mle.CommandType.CHILD_ID_RESPONSE)
-        msg.assertSentToNode(self.nodes[ROUTER2])
-        msg.assertMleMessageContainsTlv(mle.SourceAddress)
-        msg.assertMleMessageContainsTlv(mle.LeaderData)
-        msg.assertMleMessageContainsTlv(mle.Address16)
-        msg.assertMleMessageContainsOptionalTlv(mle.NetworkData)
-        msg.assertMleMessageContainsOptionalTlv(mle.Route64)
-        msg.assertMleMessageContainsOptionalTlv(mle.AddressRegistration)
-
-        # 10 - Router1
-        msg = router1_messages.next_coap_message("2.04")
-        msg.assertCoapMessageContainsTlv(network_layer.Status)
-        msg.assertCoapMessageContainsOptionalTlv(network_layer.RouterMask)
-
-        status_tlv = msg.get_coap_message_tlv(network_layer.Status)
-        self.assertEqual(network_layer.StatusValues.SUCCESS, status_tlv.status)
+
+    def verify(self, pv):
+        pkts = pv.pkts
+        pv.summary.show()
+
+        LEADER = pv.vars['LEADER']
+        ROUTER_1 = pv.vars['ROUTER_1']
+        ROUTER_2 = pv.vars['ROUTER_2']
+        MM = pv.vars['MM_PORT']
+
+        # Step 3: Verify topology is formed correctly.
+
+        pv.verify_attached('ROUTER_1')
+        _pkt_pt = pkts.filter_wpan_src64(ROUTER_1).\
+            filter_LLANMA().\
+            filter_mle_cmd(MLE_ADVERTISEMENT).\
+            must_next()
+        with pkts.save_index():
+            _pkt_id = pkts.filter_wpan_src64(ROUTER_1).\
+                filter_wpan_dst64(ROUTER_2).\
+                filter_mle_cmd(MLE_PARENT_RESPONSE).\
```

Comment:
why ROUTER_2 must attach ROUTER_1 ?
Can Router2 attach to Leader?

Reply:
This is not for checking Router2 attaching to Leader, but for checking Router1 Initial ID sequence number in Step 7

Reply:
Sorry, I misunderstood the intent.